src/mrnet/network/reaction_network.py

```diff
@@ -1541,6 +1541,44 @@ def find_paths(self, starts, target, weight, num_paths=10, ignorenode=[]):  # ->
 
         return self.PRs, paths, top_path_list
 
+    def path_analysis(self):
+        """
+        Method to rerank paths after find_paths() is perfromed.
+        """
+        heap = []
+        path_count = 0
+        for path in self.paths:
+            all_uphill_dG_full_path = 0
+            all_downhill_dG_full_path = 0
+            all_uphill_dG_path = 0
+            all_downhill_dG_path = 0
+            for step in path["full_path"]:
+                if isinstance(step, str):
+                    if self.graph.nodes[step]["free_energy"] <= 0:
+                        all_downhill_dG_full_path = all_downhill_dG_full_path + self.graph.nodes[step]["free_energy"]
+                    elif self.graph.nodes[step]["free_energy"] > 0:
+                        all_uphill_dG_full_path = all_uphill_dG_full_path + self.graph.nodes[step]["free_energy"]
+
+            for step in path["path"]:
+                if isinstance(step, str):
+                    if self.graph.nodes[step]["free_energy"] <= 0:
+                        all_downhill_dG_path = all_downhill_dG_path + self.graph.nodes[step]["free_energy"]
+                    elif self.graph.nodes[step]["free_energy"] > 0:
+                        all_uphill_dG_path = all_uphill_dG_path + self.graph.nodes[step]["free_energy"]
+            all_uphill_dG_full_path = float("{0:.4f}".format(all_uphill_dG_full_path))
+            all_uphill_dG_path = float("{0:.4f}".format(all_uphill_dG_path))
+            all_downhill_dG_full_path = float("{0:.4f}".format(all_downhill_dG_full_path))
+            all_downhill_dG_path = float("{0:.4f}".format(all_downhill_dG_path))
+            heapq.heappush(heap, (all_uphill_dG_full_path,  all_uphill_dG_path, all_downhill_dG_full_path,
+                                  all_downhill_dG_path,
+                                  len(path["path"]),
+                                  len(path["full_path"]),
+                                  path_count,path))
+            path_count = path_count + 1
+        return heap
```

Comment:
Using a heap makes sense - it's automatically sorting in place, and you can keep popping the cheapest one off the top - but isn't the most straightforward for users. Why not just return a sorted list?